tests/core/test_axes.py

```diff
@@ -58,8 +58,8 @@ def test_add_axis_on_group_entity(simulation_builder, persons, group_entity):
     simulation_builder.add_parallel_axis({'count': 2, 'name': 'rent', 'min': 0, 'max': 3000, 'period': '2018-11'})
     simulation_builder.expand_axes()
     assert simulation_builder.get_count('households') == 4
-    assert simulation_builder.get_ids('households') == ['housea0', 'houseb1', 'housea2', 'houseb3']
-    assert simulation_builder.get_input('rent', '2018-11') == approx([0, 0, 3000, 0])
+    assert simulation_builder.get_ids('households') == ['housea0', 'houseb0', 'housea1', 'houseb1']
+    assert simulation_builder.get_input('rent', '2018-11') == approx([0, 0, 3000, 3000])
```

Comment:
@benjello your insight would be appreciated. What would you expect in given this input data?

Reply:
Never used it in that case (or i do not remember). I think I would stick to your convention.

Reply:
@Morendil do you recall why you pick that expectation for `get_input('rent', '2018-11'`? Many thanks!

Reply:
It's hard to say because this was quite some time ago, but it's safest to assume that this was to reproduce the behaviour of the existing implementation, as I understood it (perhaps poorly).

The problem here is that the YAML specification for axes has lots of ambiguity. When using axes, we replicate a seed simulation N times along one or more axes. It's easy to say what this means when we have one person in the seed simulation and we say "vary this along an axis of salary between 0 and 3000", or one household and we say "vary the rent between 0 and 3000".

It becomes more complicated when there are two households, because then a "rent" value for instance is no longer a scalar but a vector. So what does it mean to say "replicate this two-household seed simulation varying the rent from 0 to 3000" ? This "0" is in fact a shorthand for the vector `[0, 0]`. But the target value is ambiguous. It could be the vector `[3000,0]` or the vector `[3000,3000]`.

As usual with OpenFisca the problem is not choosing a behaviour. There is not always a "right" or "wrong" answer and sometimes many ways could make sense. The problem is when you change something that was not formally specified but that other people have come to rely on, mentally supplying their own informal specification which is not compatible with your way. So now it works for you but it breaks for others.

So my best advice is to be careful, and to give as much explicit information as you can about why you would pick one way rather than the other. Document why you think it should work that way, with real-world examples. Compare with other users' examples.

Reply:
Maybe @adrienpacifico uses this as well and could give us some insight as how he understands the behaviour and some real-world examples?

Reply:
Would this one be considered a real-world example @Morendil ? https://github.com/openfisca/tutorial/blob/master/notebooks/how_to_handle_axes.ipynb

Reply:
@maukoquiroga Yes, insofar as it's an example of something you might actually want to calculate where axes play a role; but also no, as it's not an example of what's being discussed here i.e. how specifically we expect a seed situation with more than one household to be expanded along an axis. So you couldn't really conclude anything from that specific example that motivates a decision in this present case.